### PR TITLE
lightning: skip 'unknown database' error for schema fetching

### LINKED ISSUE #36902: Lightning: Schema Precheck Failed when DB/Tables Not Created
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

* Prepare a brand new TiDB using tiup playground
* Build a program to use the schema precheck item to check the source data

### 2. What did you expect to see? (Required)
The source data schema file can be used to get out the table structure.  The schema check can be executed based on the schema file's table structure, and report the final result. 

### 3. What did you see instead (Required)
Error occurs:
check error: cannot read schema 'XXXX' from remote: get http://127.0.0.1:10080/schema/XXXX http status code != 200, message [schema:1049]Unknown database 'XXXX'

### 4. What is your TiDB version? (Required)
Latest master version

### PR DESCRIPTION
**Pull request by dsdashun**
### What problem does this PR solve?

Issue Number: close #36902 

Problem Summary:

### What is changed and how it works?
Analyze the error message, and filter out 'Unknown database' error.  Because doing pre-check before schema being created is a reasonable use scenario.  In this case, the prechecker will use the schema file to get the table structure.  So this error can be safely ignored

### Check List

Tests 

- [x] Unit test
  - when modifying the mock implementation, the existing unit test result is also affected
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  - Create a program doing schema pre-check item on a brand new DB ( for example, start a tiup playground), and check the result
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [ ] Contains syntax changes
- [ ] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None
```

### CODE REVIEW

File: br/pkg/lightning/restore/get_pre_info.go [hunk 1]
```diff
@@ -367,7 +367,9 @@ func (p *PreRestoreInfoGetterImpl) getTableStructuresByFileMeta(ctx context.Cont
 	dbName := dbSrcFileMeta.Name
 	currentTableInfosFromDB, err := p.targetInfoGetter.FetchRemoteTableModels(ctx, dbName)
 	if err != nil {
-		return nil, errors.Trace(err)
+		if !strings.Contains(err.Error(), "Unknown database") {
```
Comment (D3Hunter) [inline br/pkg/lightning/restore/get_pre_info.go:370]: maybe use the named error https://github.com/pingcap/tidb/blob/6a6fa8a882926eb402be866fb7e35ddf20f83c42/errno/errname.go#L73

Comment (D3Hunter) [reply br/pkg/lightning/restore/get_pre_info.go:370]: maybe add it into comment

**Review by D3Hunter — APPROVED** [review]
rest lgtm

File: br/pkg/lightning/restore/get_pre_info.go (hunk 1 above)
Comment (dsdashun) [reply br/pkg/lightning/restore/get_pre_info.go]: Done in https://github.com/pingcap/tidb/pull/36904/commits/6f32ad8911de39b7ca02cd5c9e1f66b6f06fc341

**Review by buchuitoudegou — APPROVED** [review]
lgtm.

**Comment by dsdashun** [comment]
integration BR test passed: https://ci.pingcap.net/blue/organizations/jenkins/tidb_ghpr_integration_br_test/detail/tidb_ghpr_integration_br_test/10109/pipeline

**Pull request merged**